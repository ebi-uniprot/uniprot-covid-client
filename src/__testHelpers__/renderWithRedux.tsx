/* eslint-disable */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import rootReducer from '../state/rootReducer';
import { Router, Route } from 'react-router-dom';

type RenderOptions = {
  route?: string;
  history?: MemoryHistory<any>;
  path?: string;
  initialState?: any;
  store?: any;
};

const renderWithRedux = (
  ui: React.Component,
  {
    route = '',
    history = createMemoryHistory({ initialEntries: [route] }),
    path,
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  }: RenderOptions = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          {path ? <Route path={path} render={() => ui} /> : ui}
        </Router>
      </Provider>
    ),
    store,
    history,
  };
};

export default renderWithRedux;
