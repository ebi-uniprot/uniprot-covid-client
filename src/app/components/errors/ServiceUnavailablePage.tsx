import React from 'react';
import { Message } from 'franklin-sites';
import ErrorPage from './ErrorPage';
import ArtWork from '../../svg/503.svg';

const ErrorMessage = () => (
  <Message level="failure">
    <h4>This service is currently unavailable!</h4>
    <span>Please try again later</span>
  </Message>
);

const ServiceUnavailablePage = () => (
  <ErrorPage
    artwork={<ArtWork />}
    message={<ErrorMessage />}
  />
);

export default ServiceUnavailablePage;
