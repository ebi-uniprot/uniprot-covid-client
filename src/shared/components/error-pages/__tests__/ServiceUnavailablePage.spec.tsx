import React from 'react';
import ServiceUnavailablePage from '../ServiceUnavailablePage';
import renderWithRedux from '../../../../shared/__test-helpers__/RenderWithRedux';

describe('ServiceUnavailablePage component', () => {
  test('should render', () => {
    const { asFragment } = renderWithRedux(<ServiceUnavailablePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
