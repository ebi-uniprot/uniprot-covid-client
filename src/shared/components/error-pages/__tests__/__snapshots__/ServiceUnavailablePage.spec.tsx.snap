// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceUnavailablePage component should render 1`] = `
<DocumentFragment>
  <div
    class="error-page-container"
    data-testid=""
  >
    <test-file-stub
      classname="error-page-container__art-work"
    />
    <div
      class="message message--failure"
      role="status"
    >
      <div
        class="message__side-border"
      />
      <svg
        fill="currentColor"
        height="1.125em"
        stroke="currentColor"
        viewBox="0 0 19 18"
        width="1.125em"
      >
        <path
          d="M14.93 11.14h0L12.8 9l2.13-2.14a1.9 1.9 0 000-2.67h0l-.63-.64a1.9 1.9 0 00-2.68 0h0L9.5 5.7 7.35 3.55a1.9 1.9 0 00-2.68 0h0l-.63.64a1.9 1.9 0 000 2.67h0L6.18 9l-2.14 2.14a1.9 1.9 0 000 2.67h0l.63.64a1.9 1.9 0 002.68 0L9.5 12.3l2.13 2.14a1.9 1.9 0 002.68 0l.63-.64a1.9 1.9 0 000-2.67zM17.5 9a8 8 0 11-16 0 8 8 0 0116 0z"
          stroke-width="2"
        />
      </svg>
      <section
        class="message__content"
      >
        <small>
          <h4>
            This service is currently unavailable!
          </h4>
          <span>
            Please try again later
          </span>
        </small>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
